### ISSUE TITLE
refactor nomenclature for subsets and ids

### ISSUE THREAD

**Issue by randytpierce** [NOAA-GSL/VxIngest#93]
I have realized that the nomenclature surrounding subsets and ids has gotten too confusing to work with. The problem arose by the unfortunate naming of some subsets with hyphens "-" that then got translated in ids (which contain subsets) and model names (which we essentially named after subsets) to underscores "_" . It led to a mess dealing with these names in some of the code and when looking for data in the database. It became more obvious when Molly couldn't find data for the legacy subset. I decided to take a day (maybe a little more) to straiten it out in the code and in the database.

It involves changing all the references in the code, and in the json templates, shutting down the cron job, modifying the data in the database (with a N1Ql script), starting the cron job again, and then checking that everything is correct by running the tests.

Part of this effort will result in a documented naming strategy for subsets so that this doesn't happen again. Need to decide how to document this.

**Comment by randytpierce** [comment]
finished the code changes but haven't deployed it yet. Before deploying I want to have the N1QL script finalized.

**Comment by randytpierce** [comment]
Here is a reference to how to modify the key of a couchbase document. It requires a copy and delete operation.
https://forums.couchbase.com/t/change-doc-key-via-n1ql/22556
This example removes the first 4 chars in the id.
Essentially:

```
INSERT INTO Contacts (KEY _k, VALUE _v) 
SELECT SUBSTR(META(c).id,4) AS _k , c AS _v 
FROM Contacts AS c USE KEYS "leadbucket::01FD918F-2D4D-4453-A965-846914E410CC";
DELETE FROM Contacts USE KEYS  "leadbucket::01FD918F-2D4D-4453-A965-846914E410CC";
```

**Comment by randytpierce** [comment]
Using this issue for notes. This is how to transform ids ...
cbq> select case WHEN split(meta().id,':')[2]='METAR-LEGACY' THEN replace(meta().id,'METAR-LEGACY','METAR_LEGACY') WHEN split(meta().id,':')[2]='METAR-RETRO' THEN replace(meta().id,'METAR-RETRO','METAR_LEGACY_RETRO') END as newid from mdata WHERE type is not missing AND docType="ingest" AND (subset="METAR-RETRO" OR subset="METAR-LEGACY") AND version="V01";

**Comment by randytpierce** [comment]
and for subsets transforms...
cbq> select split(case WHEN split(meta().id,':')[2]='METAR-LEGACY' THEN replace(meta().id,'METAR-LEGACY','METAR_LEGACY') WHEN split(meta().id,':')[2]='METAR-RETRO' THEN replace(meta().id,'METAR-RETRO','METAR_LEGACY_RETRO') END,':')[2] as newsubset from mdata WHERE type is not missing AND docType="ingest" AND (subset="METAR-RETRO" OR subset="METAR-LEGACY") AND version="V01";

**Comment by randytpierce** [comment]
I made a backup bucket called tmp_mdata where I could copy the documents that I am going to change. This is because I have to modify the documents with the bad subset names in place to change the subset and the id fields, then copy the modified documents to new ids that are reflected by the new id field, then delete the old documents. If there is a mistake I can recover by copying all the documents in the tmp_mdata back to mdata. If successful I'll just delete tmp_mdata. These are the steps to make the change.
1) shut off the cron
2) modify the documents - ingest and data
3) modify the load_specs
4) git pull the new code
5) restart the cron.

This is how I copied the documents that need to be modified...
cbq> insert into tmp_mdata (key _k, value _v) select meta().id _k, _v from mdata _v WHERE type is not missing AND docType is not missing AND (subset="METAR-RETRO" OR subset="METAR-LEGACY") AND version="V01";
